scripts/superpmi.py

```diff
@@ -1520,6 +1554,166 @@ def print_platform_specific_environment_vars(coreclr_args, var, value):
     else:
         print("export {}={}".format(var, value))
 
+def download_index(coreclr_args):
+    """ Download the index.json for the collection.
+
+    Args:
+        coreclr_args (CoreclrArguments): parsed args
+
+    Notes:
+        The index.json file includes a dictionary of all of the different
+        collections that were done.
+    """
+
+    list_superpmi_container_uri = "https://clrjit.blob.core.windows.net/superpmi?restype=container&comp=list"
+
+    contents = urllib.request.urlopen(list_superpmi_container_uri).read().decode('utf-8')
+    urls_split = contents.split("<Url>")[1:]
+    urls = []
+    for item in urls_split:
+        url = item.split("</Url>")[0].strip()
+        
+        if "{}/{}/{}".format(coreclr_args.host_os, coreclr_args.arch, coreclr_args.build_type) in url and "index.json" in url:
+            urls.append(url)
+
+    assert(len(urls) == 1)
+    json_string = urllib.request.urlopen(urls[0]).read().decode('utf-8')
+
+    json_obj = json.loads(json_string)
+    return json_obj
+
+def download_mch(coreclr_args, specific_mch=None, include_baseline_jit=False):
+    """ Download the mch files
+
+    Args:
+        coreclr_args (CoreclrArguments): parsed args
+        specific_mch (str): Download a specific mch file
+        include_baseline_jit (bool): include downloading the baseline jit
+
+    Returns:
+        index (defaultdict(lambda: None)): collection type -> name
+    
+    """
+
+    list_superpmi_container_uri = "https://clrjit.blob.core.windows.net/superpmi?restype=container&comp=list"
```

Comment:
Looks like similar logic to above in download_index. Can they be commonized/refactored?